### ISSUE TITLE
delta delta Cq for between-sample relative quantification

### ISSUE THREAD

**Issue by ewallace** [ewallace/tidyqpcr#53]
Currently tidyqpcr calculates delta Cq for targets against reference set. We need to calculate delta delta Cq for the change in delta Cq between samples or conditions.

Needs a new `calculate_ddcq` function, or similar.

@ewallace to implement using recent reporter-gene qPCR data from Jamie Auxillos as test dataset.

**Comment by ewallace** [comment]
Created issue branch deltadeltacq-53, from the restyle branch restyle-48, for this feature.

**Comment by ewallace** [comment]
Commit 43fec3d includes a new function `calculate_deltadeltacq_bytargetid`. This runs in the new 96-well vignette, `deltacq_96well_vignette.Rmd`, the output has the right columns. We still need to test that the output values are correct, and make figures to illustrate its use.

Current function description is: 
```
#' Calculate delta delta cq (\eqn{\Delta \Delta Cq}) to globally normalize
#' quantification cycle (log2-fold) data across sample_id.
#'
#' This function does a global normalization, where all samples are compared to
#' a single sample or set of reference samples. There are other experimental
#' designs that require comparing samples in pairs or small groups, e.g. a time
#' course comparing `delta_cq` values against a reference strain at each time
#' point. For those situations, instead we recommend adapting this code to use
#' `dplyr::group_by` to draw the contrasts appropriate for the experiment.
```

It's not clear how much work we can/should do to write functions for other designs? Or does that need to be a new vignette (and new issue)?

**Comment by ewallace** [comment]
@DimmestP fixed a bug in the `calculate_deltadeltacq_bytargetid` function in #58. Branch `deltadeltacq-53` now shows and plots output from this function in the new vignette (from #51), `vignettes/deltacq_96well_vignette.Rmd`. It would be great to get a check on the vignette, which needs better explanations for `calculate_deltadeltacq_bytargetid`. 

What else do we need to check before merging into master?

**Comment by ewallace** [comment]
Closed by #62.
